src/modules/hunting/kubelet.py

```diff
@@ -208,14 +208,13 @@ def test_port_forward(self):
         # executes one command and returns output
         def test_run_container(self):
             run_url = self.path + self.Handlers.RUN.value.format(
-                podNamespace=self.pod["namespace"],
-                podID=self.pod["name"],
-                containerName=self.pod["container"],
+                podNamespace='test',
+                podID='test',
+                containerName='test',
                 cmd = ""
             )
-            status_code = requests.post(run_url, allow_redirects=False, verify=False).status_code
-            # check if return value is 4xx
-            return not 400 <= status_code < 500
+            # if we get a Method Not Allowed, we know we passed Authentication and Authorization.
+            return self.session.get(run_url, verify=False).status_code == 405
```

Comment:
I did some research and found out if we use a wrong http request method with the right url format for the endpoint, we pass Authentication and Authorization, and get a 405 (method not allowed). so checking that should be enough. wdyt? @lizrice @mcherny 

_Also I have changed the testing to use the self.session, it was a bug, now if running as a pod it uses the token._